basyx.aasservice/basyx.aasservice-backend-inmemory/src/main/java/org/eclipse/digitaltwin/basyx/aasservice/backend/InMemoryAasService.java

```diff
@@ -159,7 +159,13 @@ public File getThumbnail() {
 
 	@Override
 	public void setThumbnail(String fileName, String contentType, InputStream inputStream) {
-		String filePath = fileRepository.save(new FileMetadata(fileName, contentType, inputStream));
+		FileMetadata thumbnail = new FileMetadata(fileName, contentType, inputStream);
```

Comment:
What is the purpose of this?

Reply:
If someone tries to save a file, that already exists, an FileHandlingException is thrown in the save function of the FileRepository. If this happens in the MongoDB part (in any repsoitory), the only way to solve this I found was to delete the existing file and save the new. If this is the wrong thinking and there is a better way, I am willing to change it.